File: android-http-retrofit-rxjava3-impl/src/main/java/com/caldremch/http/exception/ApiHttpException.java
package com.caldremch.http.exception;

/**
 *
 * @author Caldremch
 * @date 2019/2/26
 * @Email caldremch@163.com
 * @describe
 *
 **/
public class ApiHttpException extends RuntimeException {

    private String mMsg;
    private int mCode;
    public ApiHttpException(int code, String message){
        super(message);
        this.mCode = code;
    }
    public int getCode() {
        return mCode;
    }
}

File: http-core/src/main/java/com/caldremch/http/core/execption/ApiHttpException.kt
package com.caldremch.http.core.execption

/**
 *
 * @author Caldremch
 * @date 2019/2/26
 * @Email caldremch@163.com
 * @describe
 */
open class ApiHttpException(val code: Int, message: String?) : RuntimeException(message)